syntax = "proto3";

package ai_service;

option go_package = "github.com/wesuuu/helpnow/backend/gen/ai_service";

service AIService {
  // Execute a specific routine for an agent
  rpc ExecuteRoutine (ExecuteRoutineRequest) returns (ExecuteRoutineResponse);
  
  // Check the status of a routine execution
  rpc GetExecutionStatus (GetExecutionStatusRequest) returns (GetExecutionStatusResponse);
  
  // Request human intervention/review status
  rpc SubmitHumanReview (SubmitHumanReviewRequest) returns (SubmitHumanReviewResponse);
}

message ExecuteRoutineRequest {
  string routine_id = 1;
  string agent_id = 2;
  map<string, string> input_params = 3;
  string user_id = 4;
}

message ExecuteRoutineResponse {
  string execution_id = 1;
  string status = 2; // "PENDING", "RUNNING", "COMPLETED", "FAILED", "WAITING_FOR_HUMAN"
}

message GetExecutionStatusRequest {
  string execution_id = 1;
}

message GetExecutionStatusResponse {
  string execution_id = 1;
  string status = 2;
  string result_data = 3; // JSON string of results
  repeated string logs = 4;
}

message SubmitHumanReviewRequest {
  string execution_id = 1;
  bool approved = 2;
  string feedback = 3;
}

message SubmitHumanReviewResponse {
  string status = 1;
}
